### PR TITLE
viewer.py: Object QA Application

### PR DESCRIPTION
**Pull request by jrreyna**
## Motivation and Context

Allows us to load in objects from datasets (like a banana, bowl, cup, etc) to simple_room.glb and see how they interact with the table geometry. The table mesh is simple and flat, so it allows us to identify any glaring problems with the geometry or physics of an object from a dataset. You can iterate through each object in the dataset, they will be displayed and rotated in front of you, then you can turn on physics and apply impulses, forces, and torques on them to see how they move. You can also make recordings of your tests by hitting the 'L' key to start and stop recording.

This is the command I use to run it:
python examples/viewer.py --scene ./data/test_assets/scenes/simple_room.glb --dataset ./data/objects/ycb/ycb.scene_dataset_config.json

After "--scene," put the .glb you want to use, and after "--dataset," put the scene_dataset_config.json you want to use. The positions of the agent and objects are hardcoded for simple_room.glb as of now

You can also just run:
python examples/viewer.py 
and it will use "./data/test_assets/scenes/simple_room.glb" and "./data/objects/ycb/ycb.scene_dataset_config.json" as defaults

Notable key bindings I added were:

'P': iterate forward through the list of dataset objects and display it
'I': iterate backward through the list of dataset objects and display it
'O': snap the existing object down to the table while turning on physics
'L': used to start and stop recording the screen
'1': apply an impulse to the object (hardcoded to be upwards and a little off its center of mass)
'2': apply a force to the object (hardcoded to be upwards and a little off its center of mass)
'3': apply an impluse torque to the object (hardcoded to be counter-clockwise)
'4': apply a torque to the object (hardcoded to be counter-clockwise)

You can see the other key bindings when you run it, as help text is printed in the terminal

## How Has This Been Tested

Nothing strenuous as of now, just gradual and periodic testing on my (John Reyna) machine as I was writing and debugging

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/viewer.py [hunk 1]
```diff
@@ -459,7 +581,145 @@ def key_press_event(self, event: Application.KeyEvent) -> None:
...
+
+        elif key == pressed.L:
+            # press L to start recording, then L again to stop it
+
+            if not self.recording and not self.writing_video:
+                # if we are not recording and not writing prev recording to file,
+                # we can start a new recording
+                self.recording = True
+                logger.info(
+                    "------------------------------------------------------------------------"
```
Comment (Skylion007) [inline examples/viewer.py:633]:
```suggestion
                    "-" * 72
```

File: examples/viewer.py [hunk 2]
```diff
@@ -23,9 +28,40 @@
...
 from habitat_sim.utils.common import quat_from_angle_axis
 
+# Setting up paths and directories for mp4 outputs
+if "google.colab" in sys.modules:
+    os.environ["IMAGEIO_FFMPEG_EXE"] = "/usr/bin/ffmpeg"
+
+repo = git.Repo(".", search_parent_directories=True)
+dir_path = repo.working_tree_dir
+# %cd $dir_path
+data_path = os.path.join(dir_path, "data")
```
Comment (ykarmesh) [inline examples/viewer.py:38]: Is this variable used somewhere?

File: examples/viewer.py [hunk 3]
```diff
@@ -268,14 +363,24 @@ def reconfigure_sim(self) -> None:
...
         # post reconfigure
         self.active_scene_graph = self.sim.get_active_scene_graph()
         self.default_agent = self.sim.get_agent(self.agent_id)
         self.agent_body_node = self.default_agent.scene_node
         self.render_camera = self.agent_body_node.node_sensor_suite.get("color_sensor")
+
         # set sim_settings scene name as actual loaded scene
         self.sim_settings["scene"] = self.sim.curr_scene_name
 
+        # Reset angent transform
```
Comment (ykarmesh) [inline examples/viewer.py:376]:
```suggestion
        # Reset agent transform
```

File: examples/viewer.py [hunk 4]
```diff
@@ -908,6 +1207,333 @@ def next_frame() -> None:
...
+
+        # change ManagedBulletRigidObject's axis of rotation
+        if self.object_rotation_axis == ObjectRotationAxis.Y:
+            self.object_rotation_axis = ObjectRotationAxis.X
+        else:
+            self.object_rotation_axis = ObjectRotationAxis.Y
+
+
+def write_video_file(self) -> None:
+    # write each PIL Image in self.video_frames to file one at a time
```
Comment (ykarmesh) [inline examples/viewer.py:1326]: Can you make the docstrings in all the functions consistent

File: examples/viewer.py [hunk 5]
```diff
@@ -787,20 +1067,36 @@ def print_help_text(self) -> None:
     SPACE:      Toggle physics simulation on/off.
     '.':        Take a single simulation step if not simulating continuously.
     'v':        (physics) Invert gravity.
+    'i':        Go backward through dataset of objects and generate it above table.
```
Comment (ykarmesh) [inline examples/viewer.py:1070]: Add instructions for video recording

File: examples/viewer.py (hunk 2 above)
Comment (jrreyna) [reply examples/viewer.py:38]: It is not 😆, good catch

File: examples/viewer.py (hunk 4 above)
Comment (jrreyna) [reply examples/viewer.py:1326]: Sorry, what do you mean by this?

Comment (ykarmesh) [reply examples/viewer.py:1326]: Yes

**Pull request closed without merging**